core/finality-grandpa/src/communication/mod.rs

```diff
@@ -273,6 +296,11 @@ impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
 		});
 
 		let topic = round_topic::<B>(round.0, set_id.0);
+
+        let mut votes_tally: HashMap<AuthorityId, VoteTally> = HashMap::new();
+		let num_signers = voters.voters().iter().count();
+		let max_timed_out = num_signers.checked_div(3).unwrap_or(1);
```

Comment:
`finality_grandpa` exposes a `fn threshold` for doing this calculation. The other thing is that voters are weighted, so we should be doing it based on total timed-out weight as opposed to a total number of voters timed-out.